example/src/pages/LinePage.tsx

```diff
@@ -132,6 +132,18 @@ const Line: React.FC<{}> = () => {
     });
   }, [state]);
 
+  const selectLine = useCallback((lineId: number) => {
+    state.forEach((_, index) => {
```

Comment:
Minor:
```suggestion
    state.some((_, index) => {
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
@JoshLipps I think we should use `forEach` as our purpose is to select the line which was clicked and deselect all other lines in the state. But with `some` we can only get Boolean value depending on the condition and the callback function will stop as soon as the condition returns `true`. And our purpose will not be fulfilled as some of the lines will not be deselected.

Reply:
ahh i see

Reply:
yes we just need to dispatch for only two lines one for selection and other for deselection, ok let me think how we implement this.

Reply:
@JoshLipps I had updated the code please review it again.

Reply:
If thats the case can we prevent the unnecessary dispatches?

Reply:
Yes I did, in current code, it only dispatch two times.